include/aws/crt/http/HttpConnectionManager.h

```diff
@@ -55,6 +58,15 @@ namespace Aws
                  * The maximum number of connections the manager is allowed to create/manage
                  */
                 size_t MaxConnections;
+
+                /** If set, initiate shutdown will return a future that will allow a user to block until the
+                 * connection manager has completely released all resources. This isn't necessary during the normal
+                 * flow of an application, but it is useful for scenarios, such as tests, that need deterministic
+                 * shutdown ordering. Be aware, if you use this anywhere other than the main thread, you will most
+                 * likely cause a deadlock. If this is set, you MUST call InitiateShutdown() before releasing your last
+                 * reference to the connection manager.
+                 */
+                bool EnableBlockingShutdown;
```

Comment:
Why not just have this be a parameter to release?

Reply:
well, the connection manager needs to know so it can accomodate the fact that the shutdown callback was installed. Also it needs to install the callback at setup time.